dlt/pipeline/pipeline.py

```diff
@@ -1372,6 +1372,13 @@ def _maybe_destination_capabilities(
             if injected_caps:
                 injected_caps.__exit__(None, None, None)
 
+    def _normalize_dataset_name(self, dataset_name: str) -> str:
+        """Normalizes the dataset name using the dataset_name_layout."""
+        if "%s" not in self.config.dataset_name_layout:
```

Comment:
please move this check to `on_resolved` method of `PipelineConfiguration`. this method is called after all fields were successfully resolved by config providers and is used to do additional processing and checks.

Raise `ConfigurationValueError` (specialized ValueError), you can also inspect other `on_resolved` impls to see what we typically do there

Reply:
> return self.config.dataset_name_layout % dataset_name
this line is certainly correct

but the validation line
> if "%s" not in self.config.dataset_name_layout:
could be moved to `on_resolved` because it depends only on config instance content.

ping me on Slack if this is not clear... maybe I'm missing some nuances :)